### DISCUSSION TITLE
Audio stutter and buffer underrun errors.

### DISCUSSION THREAD

**Discussion by loughkb** [AlexandreRouma/SDRPlusPlus#295] [General]
I'm running POP! OS on my System76 laptop.  Based on Ubuntu 20.04.
I tried installing from the proper .deb file and also tried building from source.
The audio stutters badly and the terminal fills with 'audio write error:  buffer underrun'.
Changing sample rate for the audio or switching the device and switching it back will *sometimes* get me clean audio and no errors.  As soon as I change frequency or mode, anything that causes a short pause in the audio, it will start stuttering again and spitting out the underrun errors.
Also, I'd love to see 'pulseaudio' added as an option.   It lists my audio device 14 times but always gives an error about the device being in use, I have to select 'default' to get audio.  Then I have the stuttering and underrun errors.

**Comment by AlexandreRouma** [comment]
Hello, this is a common issue with RtAudio on debian based systems, please check https://github.com/AlexandreRouma/SDRPlusPlus/discussions/112

**Reply by loughkb** [reply]
Thank you!!!!    I actually just found that and rebuilt rtaudio as instructed.   Fixed it!
Now if I could just get it to work with my SDRPlay RSPdx.  It works fine with my airspy.
With the sdrplay and the direct support selected, I just get a solid yellow waterfall with one peak in the middle regardless of frequency.  If I use it through the SoapySDR module, it looks like it receives, but no signals, regardless of antenna input selection or gain settings.

**Reply by AlexandreRouma** [reply]
Could you show a screenshot of the full UI with the SDRplay source?
